src/window/window_wrapper/mod.rs

```diff
@@ -162,41 +164,44 @@ impl GlutinWindowWrapper {
 
     pub fn draw_frame(&mut self, dt: f32) {
         let window = self.windowed_context.window();
-        let current_size = window.inner_size();
-        let previous_size = self.saved_inner_size;
-
-        if previous_size != current_size {
-            trace!("Updating grid size: {:#?}", current_size);
-            self.saved_inner_size = current_size;
-            self.handle_new_grid_size(current_size);
-            self.skia_renderer.resize(&self.windowed_context);
-        }
 
         if REDRAW_SCHEDULER.should_draw() || SETTINGS.get::<WindowSettings>().no_idle {
-            let renderer = &mut self.renderer;
+            self.renderer.draw_frame(self.skia_renderer.canvas(), dt);
+            self.skia_renderer.gr_context.flush(None);
+            self.windowed_context.swap_buffers().unwrap();
+        }
 
-            {
-                let canvas = self.skia_renderer.canvas();
+        // Wait until fonts are loaded, so we can set proper window size.
+        if !self.renderer.is_ready {
+            return;
+        }
 
-                if renderer.draw_frame(canvas, dt) {
-                    self.handle_new_grid_size(current_size);
-                }
+        if self.first_render_pass {
+            self.first_render_pass = false;
+            if !window.is_maximized() {
+                window.set_inner_size(self.renderer.to_physical_size(self.saved_grid_size));
             }
+        }
 
-            self.skia_renderer.gr_context.flush(None);
-            self.windowed_context.swap_buffers().unwrap();
+        let new_size = window.inner_size();
+
+        if self.saved_inner_size != new_size {
+            self.saved_inner_size = new_size;
+            self.handle_new_grid_size(new_size);
+            self.skia_renderer.resize(&self.windowed_context);
         }
     }
 
-    fn handle_new_grid_size(&self, new_size: PhysicalSize<u32>) {
-        if new_size.width > 0 && new_size.height > 0 {
-            // Add 1 here to make sure resizing doesn't change the grid size on startup
-            let width = ((new_size.width + 1) / self.renderer.font_width as u32) as u32;
-            let height = ((new_size.height + 1) / self.renderer.font_height as u32) as u32;
-            self.ui_command_sender
-                .send(UiCommand::Resize { width, height })
-                .ok();
+    fn handle_new_grid_size(&mut self, new_size: PhysicalSize<u32>) {
+        let (width, height) = self.renderer.to_grid_size(new_size);
```

Comment:
I was thinking, maybe we can leave only saved_grid_size and remove saved_inner_size, but my attempts to do so failed.

So i'm leaving both for now. Maybe we come up with better ideas later :)